### DISCUSSION TITLE
Decoding ids

### DISCUSSION THREAD

**Discussion by d3bt3ch** [sqids/sqids-java#19] [General]
From the FAQ `decoding random IDs can sometimes produce the same numbers`. What are the odds, because we are planning to use `sqids` to stringify our long ids in json output. If two ids decodes to the same long then it could be a problem.

**Comment by 0x3333** [comment]
If the IDs you generated are using the same parameters, they can't collide, they are unique.

The FAQ is regarding a random generated ID, like brute-force, etc. The algorithm is not cryptographically secure.

**Comment by 4kimov** [comment]
The odds don't really matter, because you should always check incoming IDs by re-encoding decoded numbers to see if original ID matches. That's how you would know if it's valid or not. https://sqids.org/faq#valid-ids
